packages/react-conformance/src/utils/getCallbackArguments.test.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * @jest-environment node
+ */
+
+import { DirectoryJSON, fs as memFS, vol } from 'memfs';
+import * as realFS from 'fs';
+import * as ts from 'typescript';
+
+import { getCallbackArguments } from './getCallbackArguments';
+
+/**
+ * Creates a CompilerHost that reads files from memfs.
+ */
+function createCompilerHost(): ts.CompilerHost {
+  const originalHost = ts.createCompilerHost({});
+
+  function fileExists(fileName: string): boolean {
+    return memFS.existsSync(fileName) || realFS.existsSync(fileName);
+  }
+
+  function getSourceFile(fileName: string, languageVersion: ts.ScriptTarget) {
+    let sourceText;
+
+    if (memFS.existsSync(fileName)) {
+      sourceText = memFS.readFileSync(fileName, 'utf8') as string;
+    } else if (realFS.existsSync(fileName)) {
+      sourceText = realFS.readFileSync(fileName, 'utf8');
+    }
+
+    return sourceText === undefined ? undefined : ts.createSourceFile(fileName, sourceText, languageVersion);
+  }
+
+  return {
+    ...originalHost,
+
+    readFile: () => {
+      throw new Error('Not supported');
+    },
+    writeFile: () => {
+      throw new Error('Not supported');
+    },
+
+    getSourceFile,
+    fileExists,
+  };
+}
+
+async function setupProgram(
+  rootNames: ts.CreateProgramOptions['rootNames'],
+  directoryJSON: DirectoryJSON,
+): Promise<ts.Program> {
+  vol.fromJSON(directoryJSON);
+
+  return ts.createProgram({
+    rootNames,
+    options: {},
+    host: createCompilerHost(),
+  });
+}
+
+describe('getCallbackArguments', () => {
+  describe('params', () => {
+    it('handles "null" as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (e: null) => void; }',
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: null });
+    });
+
+    it('handles primitives as a param', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `export interface AccordionProps {
+          onToggle: (a: string, b: number, c: boolean, d: undefined) => void;
+        }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ a: 'String', b: 'Number', c: 'Boolean', d: undefined });
+    });
+
+    it('handles arrays', async () => {
```

Comment:
nit: it kind of seems like a lot of copy pasted test structure, since you always provide the same structure of input and assertion, it would be nice to use parametrized test like `it.each`

Reply:
I agree, but due size of fixtures I would like to leave them as it is. It also easier to use `it.only` to debug failing tests